docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,324 @@
+#endif
+#endif
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	MPU_REG_NUM_APP_TXT,		/* Apps */
+	MPU_REG_NUM_APP_RO,
+	MPU_REG_NUM_APP_DATA,
+#else
+	MPU_REG_NUM_APP,
+#endif
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+	MPU_REG_NUM_STK,		/* Stack */
+#endif
+#else
+	MPU_REG_NUM_UFLASH,
+	MPU_REG_NUM_URAM,
+#endif
+	MPU_REG_NUM_MAX
+};
+```
+
+## How to Control MPU regions in TizenRT
+### Add a new MPU region
+1. Add your MPU region in enum MPU_REG_NUM in mpu.h file.
+2. Call function mpu_get_register_value to configure your MPU region with appropriate arguments.
+	- If (uint32_t \*)regs argument is passed as NULL, MPU region settings will be directly written into MPU registers
+	- If instead a variable (like uint32_t var[3]) is passed as argument, then MPU register values are saved in this variable and can be used at appropriate place for setting MPU region (ex. during context switch).
+
+### Delete a MPU region
+1. Remove MPU region from enum MPU_REG_NUM in mpu.h file.
+2. Remove calls made to mpu_get_register_value using this MPU region as the (uint32_t) region argument.
+
+### Modify a MPU region
+1. For overlapping memory address range between 2 MPU regions, modify their relative position in enum MPU_REG_NUM as per below rule:
+	- Attributes of higher MPU region number will override those set by lower numbered MPU region for this overlapping address range.
+
+## Current Usage of MPU regions in TizenRT
+- In Protected Build, kernel runs in privileged mode. It has access to all memory ranges and it defines memory access permissions for unprivileged user code using MPU regions.
+
+- Below 2 cases arise regarding user code for Protected build:
+### 1. CONFIG_APP_BINARY_SEPARATION=n
+- This means there is a single user binary.
+- In this case, Access and execute permissions are set as below for MPU regions:
+
+```
+	--------------------------------------------------------------------
+	MPU Region		Memory Segments		Permissions
+	--------------------------------------------------------------------
+	MPU_REG_NUM_UFLASH	TEXT			ROX
+	MPU_REG_NUM_URAM	DATA/HEAP/STACK		RWNX
+	--------------------------------------------------------------------
+```
+
+- The functions **mpu_userflash** and **mpu_userintsram** are used to configure MPU regions for User Flash and User RAM respectively
+
+### 2. CONFIG_APP_BINARY_SEPARATION=y
+- This means there are multiple user binaries.
+- Each of these binaries has it's own memory address range in RAM.
+- Separate access permissions, using MPU regions, need to be defined for each binary.
+
+#### Configuring MPU regions for ELF App
+- In TizenRT, each app is generated as a separate ELF binary.
+- Dynamic ELF loader will load each app's TEXT, RO, DATA etc. segments to RAM during boot-up.
+- Settings for an MPU region dedicated for apps are saved in the app's TCB structure (struct tcb_s \*) inside mpu_regs[] member.
+- These settings are assigned and saved, inside exec_module function, using mpu_get_register_value function.
```

Comment:
These settings are assigned and saved ===>> The contents of mpu_regs[] is initialized inside ....

Reply:
done